crates/kernel/src/tasks/scheduler.rs

```diff
@@ -232,18 +259,26 @@ impl Scheduler {
         // the given input, clearing the input request out.
         trace!(?input_request_id, ?input, "Received input for task");
 
-        let mut inner = self.inner.write().unwrap();
-        let Some(task_id) = inner.input_requests.get(&input_request_id) else {
+        let mut inner = self.task_q.lock().unwrap();
+
+        // Find the task that requested this input, if any
+        let Some((task_id, perms)) = inner.suspended.iter().find_map(|(task_id, sr)| {
```

Comment:
Is it ever, even including all imaginable future changes, possible for multiple tasks to wait for the same input?

Reply:
It should not be, at least not in the LambdaMOO model of networking.